buffer/src/main/java/io/netty/buffer/api/bytebuffer/NioBuffer.java

```diff
@@ -192,6 +191,42 @@ public void copyInto(int srcPos, Buffer dest, int destPos, int length) {
         Statics.copyToViaReverseLoop(this, srcPos, dest, destPos, length);
     }
 
+    @Override
+    public int bytesBefore(byte needle) {
+        if (!isAccessible()) {
+            throw bufferIsClosed(this);
+        }
+        int offset = roff;
+        final int length = woff - roff;
+        final int end = woff;
+
+        if (length > 7) {
+            final long pattern = (needle & 0xFFL) * 0x101010101010101L;
```

Comment:
I would add some comments on the source of this algo (can be either the hackers's delight chapter or richard s blog article)